python/tests/test_table_read.py

```diff
@@ -160,6 +160,21 @@ def test_read_table_with_column_subset():
     )
 
 
+def test_read_table_as_category():
```

Comment:
We should make sure this new code path works if there are partitioning columns. I think as it is written it likely won't.